src/compat/dbtype.h

```diff
@@ -791,6 +791,9 @@ union db_char
     unsigned char codeset;
     int size;
     char *buf;
+    int compressed_length;
+    char *compressed_buf;
+    char was_compressed;	/* Compression status */
```

Comment:
The name is a bit confusing, I was tempted to say it is redundant, but I think it means that the compression was attempted (with success or not) before (when compressed_buf == NULL).
What about renaming it or replacing it with a flag 'uncompressable" which is to be set only if compression was attempted and failed ?